homeassistant/components/zha/__init__.py

```diff
@@ -73,45 +65,91 @@ class RadioType(enum.Enum):
 
 # ZigBee definitions
 CENTICELSIUS = 'C-100'
-# Key in hass.data dict containing discovery info
-DISCOVERY_KEY = 'zha_discovery_info'
 
 # Internal definitions
 APPLICATION_CONTROLLER = None
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup(hass, config):
+    """Set up ZHA from config."""
+    if DOMAIN in config:
+        if not hass.config_entries.async_entries(DOMAIN):
+            hass.async_add_job(hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={'source': config_entries.SOURCE_IMPORT},
+                data=config[DOMAIN]
+            ))
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
     """Set up ZHA.
 
     Will automatically load components to support devices found on the network.
     """
     global APPLICATION_CONTROLLER
```

Comment:
Don't use global. Store this in hass.data instead.

Reply:
This comes from the initial implementation, if possible I would like to leave this for now and fix it in a separate PR.